ovs/extensions/healthcheck/openvstorage/openvstoragecluster_health_check.py

```diff
@@ -170,91 +162,92 @@ def _list_dirs_in_directory(pwd):
-
-        :param logger: logging object
-        :type logger: ovs.log.healthcheck_logHandler.HCLogHandler
-        :param process_name: name of a certain process running on this local machine
-        :type process_name: str
-        :param port: port where the service is running on
-        :type port: int
-        :param ip: ip address to try
-        :type ip: str
-        """
-        logger.info('Checking port {0} of service {1} ...'.format(port, process_name), '_is_port_listening')
-        if Helper.check_port_connection(port, ip):
-            logger.success('Connection successfully established!',
-                           'port_{0}_{1}'.format(process_name, port))
-        else:
-            logger.failure('Connection FAILED to service {1} on port {0}'.format(port, process_name),
-                           'port_{0}_{1}'.format(process_name, port))
+        Checks the extra ports from ovs
+        :param result_handler:
+        :return:
+        """
+        return OpenvStorageHealthCheck._check_extra_ports(result_handler, 'nginx')
 
     @staticmethod
-    @expose_to_cli('ovs', 'required-ports-test')
-    def check_required_ports(logger):
+    @expose_to_cli('ovs', 'memcached-ports-test')
+    def check_memcached_ports(result_handler):
         """
-        Checks all ports of Open vStorage components (framework, memcached, nginx, rabbitMQ and celery)
-
-        :param logger: logging object
-        :type logger: ovs.log.healthcheck_logHandler.HCLogHandler
+        Checks the extra ports from ovs
+        :param result_handler:
+        :return:
         """
+        return OpenvStorageHealthCheck._check_extra_ports(result_handler, 'memcached')
 
-        logger.info('Checking PORT CONNECTIONS of OVS & EXTRA services ...', '')
-
-        # check ports for OVS services
-        logger.info('Checking OVS ports', '')
-        for sr in ServiceHelper.get_services():
-            if sr.storagerouter_guid == OpenvStorageHealthCheck.LOCAL_SR.guid:
-                for port in sr.ports:
-                    if sr.name.split('_')[0] == 'albaproxy':
-                        storagedriver_id = '{0}{1}'.format(sr.name.split('_')[1], OpenvStorageHealthCheck.LOCAL_ID)
-                        ip = StoragedriverHelper.get_by_storagedriver_id(storagedriver_id).storage_ip
-                        OpenvStorageHealthCheck._is_port_listening(logger, sr.name, port, ip)
-                    else:
-                        OpenvStorageHealthCheck._is_port_listening(logger, sr.name, port)
-
-        # check NGINX and memcached
-        logger.info('Checking EXTRA ports', '')
-        for process, ports in Helper.extra_ports.iteritems():
+    @staticmethod
+    def _check_extra_ports(result_handler, key):
+        """
+        Checks the extra ports specified in the settings.json
+        :param result_handler:
+        :return:
+        """
+        result_handler.info('Checking {0} ports'.format(key))
+        for ports in Helper.extra_ports[key]:
```

Comment:
Why retrieve these ports from the ```settings.json```
For nginx we don't store this anywhere, but memcached ports can be retrieved from the configuration management